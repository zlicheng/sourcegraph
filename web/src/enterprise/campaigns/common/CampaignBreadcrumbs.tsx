import React from 'react'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { Link } from '../../../../../shared/src/components/Link'

interface Props {
    campaign: Pick<GQL.ICampaign, 'name' | 'url'> | null
    className?: string
}

export const CampaignBreadcrumbs: React.FunctionComponent<Props> = ({ campaign, className = '' }) => (
    <nav aria-label="breadcrumb" className={className}>
        <ol className="breadcrumb">
            {campaign ? (
                <>
                    <li className="breadcrumb-item">
                        <Link to="/campaigns">Campaigns</Link>
                    </li>
                    <li className="breadcrumb-item active">
                        <Link to={campaign.url}>{campaign.name}</Link>
                    </li>
                </>
            ) : (
                <li className="breadcrumb-item active" aria-current="page">
                    Campaigns
                </li>
            )}
        </ol>
    </nav>
)
