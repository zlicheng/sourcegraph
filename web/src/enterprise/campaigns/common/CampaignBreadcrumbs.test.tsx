import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { CampaignBreadcrumbs } from './CampaignBreadcrumbs'

describe('CampaignBreadcrumbs', () => {
    test('new', () => expect(createRenderer().render(<CampaignBreadcrumbs />)).toMatchSnapshot())

    test('existing', () =>
        expect(
            createRenderer().render(
                <CampaignBreadcrumbs campaign={{ name: 'My campaign', url: 'https://example.com' }} />
            )
        ).toMatchSnapshot())
})
