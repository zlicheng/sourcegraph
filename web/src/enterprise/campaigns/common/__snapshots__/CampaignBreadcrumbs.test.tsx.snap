// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignBreadcrumbs all changesets not open 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs closed 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs edit mode 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs editable 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs existing 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <li
      aria-current="page"
      className="breadcrumb-item active"
    >
      Campaigns
    </li>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs mode: closing 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs mode: deleting 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs mode: saving 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs new 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <li
      aria-current="page"
      className="breadcrumb-item active"
    >
      Campaigns
    </li>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs not editable 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs processing 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;

exports[`CampaignBreadcrumbs some changesets still open 1`] = `
<nav
  aria-label="breadcrumb"
>
  <ol
    className="breadcrumb"
  >
    <React.Fragment>
      <li
        className="breadcrumb-item"
      >
        <AnchorLink
          to="/campaigns"
        >
          Campaigns
        </AnchorLink>
      </li>
      <li
        className="breadcrumb-item active"
      >
        <AnchorLink>
          Super campaign
        </AnchorLink>
      </li>
    </React.Fragment>
  </ol>
</nav>
`;
